pkg/types/types.go

```diff
@@ -68,8 +115,11 @@ type ResourceFactory interface {
 	GetResourceServer(ResourcePool) (ResourceServer, error)
 	GetInfoProvider(string) DeviceInfoProvider
 	GetSelector(string, []string) (DeviceSelector, error)
-	GetResourcePool(rc *ResourceConfig, deviceList []PciNetDevice) (ResourcePool, error)
+	GetResourcePool(rc *ResourceConfig, deviceList []PciDevice) (ResourcePool, error)
 	GetRdmaSpec(string) RdmaSpec
+	GetDeviceProvider(DeviceType) DeviceProvider
+	// GetDeviceFilter(*ResourceConfig) (DeviceFilter, error)
+	GetDeviceFilter(*ResourceConfig) (interface{}, error)
```

Comment:
Seems like an interface definition is missing here. i dont understand from the definition here what should GetDeviceFilter return value represents

looking at the code i get that its an opaque to carry the parsed selector to the device provider.

Reply:
Well, interface{} is like a void pointer in C to which you can return any struct. This makes it generic method to return different selector struct. I will remove the commented line above(old definition)

Reply:
Perhaps explain as well what GetDeviceFilter is expected to do/return.